wbooks/src/App/Main/index.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+import Dashboard from './Dashboard';
+import BookDetail from './BookDetail';
+import Profile from './Profile';
+import './styles.css';
+
+class Main extends React.Component {
+  render() {
+    return(
+      <div className="main-content">
+        <Switch>
+          <Route exact path='/' render={() => <Dashboard books={this.props.books} />} />
+          <Route path='/books/:number' render={({match}) => <BookDetail books={this.props.books} match={match} />}/>
```

Comment:
Maybe name the param `:id` instead of `:number`?

Also, here since you are using a function to render the route, you can pass the id directly like: `id={match.id}`.